Refactor code for better architecture on API (HMI interface). (#44)
IRT-Saint-Exupery/CoFMPy: docs/user_guide/CofmuPy_gui.md

# 🚀 Using Graphical user interface

> Graphical user interface is a web app, executed throw web browser.

Reviewer: ```suggestion
Graphical user interface is a web app, executed through web browser.
```

> Once installed and started, it works locally and is accessible on url http://localhost:5000

Reviewer: ```suggestion
Once installed and started, it works locally and is accessible at the URL http://localhost:5000
```

## 📚 Application content

Application is composed of 2 main pages :
- `Projects management` page, to manage available use cases list.
    * Create, edit and delete use cases
    * Load Project => switch to `Project edit` page below

![GUI_Project.png](../assets/gui/GUI_Project.png)

- `Project edit` page composed of different sections :
    * General cosimulation information
    * FMUs information

>       * Add fmus

Reviewer: ```suggestion
      * Add FMUs
```

      * Edit FMU caracteristics and initializations

>     * Connections to manage links between fmus

Reviewer: ```suggestion
    * Connections to manage links between FMUs
```

![GUI_Edit_Project.png](../assets/gui/GUI_Edit_Project.png)

## 🐍 User guide - Workflow

> HMI is design to be very simple and intuitive, follow steps below to obtain configuration file and execute cosimulation with CoFmuPy.

Reviewer: ```suggestion
HMI is designed to be very simple and intuitive, follow steps below to obtain configuration file and execute cosimulation with CoFmuPy.
```

### 1. Create use case
![create_use_case](../assets/gui/create_use_case.png)  

### 2. Empty page after use creation
![empty_page](../assets/gui/empty_page.png)

### 3. Select simulation options
![simulation_options](../assets/gui/simulation_options.png)

### 4. Add fmu by selecting `+` button
![add_fmu](../assets/gui/add_fmu.png)

### 5. Select fmu file (or drop), edit characteristics and clic on upload fmu button
![upload_fmu](../assets/gui/upload_fmu.png)  

> Repeat operations 4 and 5 for each expected fmu

Reviewer: ```suggestion
Repeat operations 4 and 5 for each expected FMU
```
